packages/eas-cli/src/commands/update/index.ts

```diff
@@ -600,15 +630,69 @@ async function getRuntimeVersionObjectAsync(
     }
   }
 
-  return Object.fromEntries(
-    platforms.map(platform => [
-      platform,
-      nullthrows(
-        Updates.getRuntimeVersion(exp, platform),
-        `Unable to determine runtime version for ${platform}`
-      ),
-    ])
-  );
+  try {
+    return [transformRuntimeVersions(exp, platforms), exp];
+  } catch (error: any) {
+    if (nonInteractive) {
+      throw error;
+    }
+
+    Log.fail(error.message);
+
+    const runConfig = await selectAsync(
+      `Do you want us to run automatic ${chalk.bold(
+        'app.json'
+      )} configuration for EAS Update for you?`,
+      [
+        { title: 'Yes', value: true },
+        {
+          title: 'No, I will set the runtime version manually (EAS CLI exits)',
+          value: false,
+        },
+      ]
+    );
+
+    if (!runConfig) {
+      Errors.exit(1);
+    }
+
+    const workflows = await resolveWorkflowPerPlatformAsync(projectDir);
+    const configUpdate = await configureAppJSONForEASUpdateAsync({
+      exp,
+      projectDir,
+      projectId,
+      platform: platformFlag as RequestedPlatform,
+      workflows,
+    });
+
+    const newConfig: ExpoConfig = { ...exp, ...configUpdate };
+
+    await configureNativeFilesForEASUpdateAsync({
+      exp: newConfig,
+      projectDir,
+      projectId,
+      platform: platformFlag as RequestedPlatform,
+      workflows,
+      graphqlClient,
+    });
+
+    const continueWithChanges = await selectAsync(
+      `Would you like to continue update process with uncommitted changes in repository?`,
+      [
+        { title: 'Yes', value: true },
+        {
+          title: 'No, I will commit the modified files first (EAS CLI exits)',
```

Comment:
```suggestion
          title: 'No, I will commit the modified files first',
```

Reply:
This pattern of adding `(EAS CLI exits)` at the end of exiting option has been taken from the existing code.

I we want to remove it, I feel like this should be done in the whole codebase, in the separate PR, but personally I like this addition, and I think that make it clear for the user what will happed.

Reply:
It looks like there is only one other place where we do this and it's in a fairly obscure place: [https://github.com/expo/eas-cli/blob/2aade4c0c82665f12cd8053d339122fb35f56168/packages/eas-cli/src/commands/build/index.ts#L305](https://github.com/expo/eas-cli/blob/2aade4c0c82665f12cd8053d339122fb35f56168/packages/eas-cli/src/commands/build/index.ts#L305)



I'm not entirely opposed to letting users know that the process will exit but I found this wording to be confusinghttps://github.com/expo/eas-cli/blob/2aade4c0c82665f12cd8053d339122fb35f56168/packages/eas-cli/src/commands/build/index.ts#L305. It'd be better to say something like (this process will exit)